### PR TITLE
Set up throttling for two factor login

### PR DESCRIPTION
**Pull request by zzuzzy**

### CODE REVIEW

File: security/throttling.py [hunk 1]
```diff
@@ -43,27 +43,43 @@ def _validate(self, request):
...
         return count_same_requests <= self.throttle_at
 
 
-class SuccessfulLoginThrottlingValidator(ThrottlingValidator):
+class UnsuccessfulLoginThrottlingValidator(LoginThrottlingValidator):
+
+    type = InputLoggedRequest.UNSUCCESSFUL_LOGIN_REQUEST
+
+    def __init__(self, timeframe, throttle_at, description=_('Too many login attempts')):
+        super(UnsuccessfulLoginThrottlingValidator, self).__init__(timeframe, throttle_at, description)
```
Comment (rubickcz) [inline security/throttling.py]: This is old style syntax, use just `super()`.

File: security/throttling.py [hunk 2]
```diff
@@ -43,27 +43,43 @@ def _validate(self, request):
...
+
+
+class SuccessfulLoginThrottlingValidator(LoginThrottlingValidator):
+
+    type = InputLoggedRequest.SUCCESSFUL_LOGIN_REQUEST
 
     def __init__(self, timeframe, throttle_at, description=_('You are logged too much times')):
```
Comment (rubickcz) [inline security/throttling.py]: Maybe better sentence would be "You have logged in too many times".

File: security/throttling.py [hunk 3]
```diff
@@ -43,27 +43,43 @@ def _validate(self, request):
...
+
+    def __init__(self, timeframe, throttle_at, description=_('Too many login attempts')):
+        super().__init__(timeframe, throttle_at, description)
+
+
+class SuccessfulTwoFactorCodeVerificationThrottlingValidator(LoginThrottlingValidator):
+
+    type = InputLoggedRequest.SUCCESSFUL_2FA_CODE_VERIFICATION_REQUEST
+
+    def __init__(self, timeframe, throttle_at, description=_('You are logged too much times')):
```
Comment (rubickcz) [inline security/throttling.py]: A better sentence would be: "You have logged in too many times"

File: security/locale/cs/LC_MESSAGES/django.po [hunk 4]
```diff
@@ -18,203 +18,211 @@ msgstr ""
 "Content-Transfer-Encoding: 8bit\n"
 "Plural-Forms: nplurals=3; plural=(n==1) ? 0 : (n>=2 && n<=4) ? 1 : 2;\n"
 
-#: cores.py:32 models.py:129 models.py:160
+#: cores.py:30 models.py:168 models.py:199
```
Comment (rubickcz) [inline security/locale/cs/LC_MESSAGES/django.po]: ouch, we should use `--no-location` when generating this file.

Comment (rubickcz) [reply security/locale/cs/LC_MESSAGES/django.po]: @zzuzzy can you please do it with that option?

Comment (zzuzzy) [reply security/locale/cs/LC_MESSAGES/django.po]: Ok, location is now removed.

File: security/throttling.py [hunk 5]
```diff
@@ -43,27 +43,43 @@ def _validate(self, request):
...
-    def __init__(self, timeframe, throttle_at, description=_('Too many login attempts')):
-        super(UnsuccessfulLoginThrottlingValidator, self).__init__(timeframe, throttle_at, description)
+class LoginThrottlingValidator(ThrottlingValidator):
 
     def _validate(self, request):
         count_same_requests = InputLoggedRequest.objects.filter(
             ip=get_ip(request), path=request.path,
             request_timestamp__gte=timezone.now() - timedelta(seconds=self.timeframe),
-            type=InputLoggedRequest.UNSUCCESSFUL_LOGIN_REQUEST).count()
+            type=self.type).count()
```
Comment (matllubos) [inline security/throttling.py]: proč to tady filtrujeme podle typu? Já bych to nedělal je to zbytečné. Ten typ tam dáváme spíš pro administrátorské účely

**Pull request merged**